### ISSUE TITLE
Installation issue: pegtl with intel compiler (as dependency of paraview)

### ISSUE THREAD

**Issue by EthanS94** [LLNL/spack#19313]
### Steps to reproduce the issue

```console
$ spack install pegtl@2.8.3%intel@19.0.4.243 build_type=RelWithDebInfo arch=linux-rhel7-broadwell
...
cd /tmp/stam/spack-stage/spack-stage-pegtl-2.8.3-s663irnyp7pjq4alxlipn5x7tp3274hw/spack-build-s663irn/src/test/pegtl && /lustre/scratch4/yellow/.mdt2/stam/pilotlight-spack-snow-install/linux-rhel7-broadwell/intel-19.0.4.243/cmake-3.15.3-j3tge5othyfgmezm7jphaurp5d4riwwm/bin/cmake -E cmake_link_script CMakeFiles/pegtl-test-pegtl_string_t.dir/link.txt --verbose=1
/yellow/usr/projects/paraview/stam/pilotlight-dev/tests/spack/tmp_snow_develop_intel_19.0.4.243/spack/lib/spack/env/intel/icpc  -O2 -g -DNDEBUG   CMakeFiles/pegtl-test-pegtl_string_t.dir/pegtl_string_t.cpp.o  -o pegtl-test-pegtl_string_t 
/tmp/stam/spack-stage/spack-stage-pegtl-2.8.3-s663irnyp7pjq4alxlipn5x7tp3274hw/spack-src/include/tao/pegtl/contrib/rep_one_min_max.hpp(33): error #186: pointless comparison of unsigned integer with zero
                 if( size < Min ) {
                          ^
```

### Information on your system
* **Spack:** 0.15.4-1467-71d2ce740
* **Python:** 3.6.3
* **Platform:** linux-rhel7-broadwell

* [compilers.yaml](https://github.com/spack/spack/files/5380376/compilers.yaml.txt)
* [packages.yaml](https://github.com/spack/spack/files/5380378/packages.yaml.txt)

### Additional information

* [spack-build-out.txt]()
* [spack-build-env.txt]()

### General information

- [x] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [x] I have run `spack maintainers <name-of-the-package>` and @mentioned any maintainers @chuckatkins 
- [ ] I have uploaded the build log and environment files
- [x] I have searched the issues of this repo and believe this is not a duplicate

**Comment by DarylGrunau** [comment]
This issue should be resolved by PR #21866 ~when you spec `pegtl~tests~examples`~.

**Comment by alalazo** [comment]
#21866 has been merged
